remoulade/api/main.py

```diff
@@ -39,27 +29,24 @@ class GroupMessagesT(TypedDict):
     messages: List[dict]
 
 
-@app.route("/messages/states")
+@app.route("/messages/states", methods=["POST"])
 def get_states():
-    args = PageSchema().load(request.args.to_dict())
-    backend = remoulade.get_broker().get_state_backend()
-    data = [s.as_dict() for s in backend.get_states()]
-    if args.get("search_value"):
-        keys = ["message_id", "name", "actor_name", "args", "kwargs", "options"]
-        value = args["search_value"].lower()
-        data = [item for item in data if dict_has(item, keys, value)]
+    states_kwargs = PageSchema().load(request.json or {})
+    backend = get_broker().get_state_backend()
+    data = [state.as_dict() for state in backend.get_states(**states_kwargs)]
+    return {"data": data, "count": len(data)}
 
-    if args.get("sort_column"):
-        reverse = args.get("sort_direction") == "desc"
-        sort_column = args["sort_column"]
-        data = sort_dicts(data, sort_column, reverse)
 
-    return {"data": data[args["offset"] : args["size"] + args["offset"]], "count": len(data)}
+@app.route("/messages/states", methods=["DELETE"])
+def clean_states():
+    states_kwargs = DeleteSchema().load(request.json or {})
+    get_broker().get_state_backend().clean(**states_kwargs)
```

Comment:
this only works for postgres, check the backend used before and return an error if it is not pg